src/OpenTK/Platform/X11/X11GLContext.cs

```diff
@@ -385,15 +385,14 @@ public override int SwapInterval
                         Glx.QueryDrawable(Display, currentWindow.Handle, GLXAttribute.SWAP_INTERVAL_EXT, out value);
                         return value;
                     }
-                    else if (vsync_mesa_supported)
+                    if (vsync_mesa_supported)
                     {
                         return Glx.Mesa.GetSwapInterval();
                     }
-                    else if (vsync_sgi_supported)
+                    if (vsync_sgi_supported)
                     {
                         return sgi_swap_interval;
                     }
```

Comment:
Why should this no longer be an else-if chain?

Reply:
Somewhat depends on the style you want here, both are exactly the same in practice.
This way is IMHO cleaner (and is what ReSharper prefers)

Thoughts FWIW:
You first should check for the XExtension version of vsync, as this is the newest / most common method of doing things.
Checking for the mesa vysnc support is a reasonable fallback.
SGI_SWAPINTERVAL is mega old, and quite often broken, and should be ignored unless we have anything else.

Reply:
I think this should still be an else-if chain to make clear that two of these should not be true at the same time. I would be confused by this code if it was written without the else-if chaining.